### PR TITLE
Use flipper-activerecord with AR3.2

### PR DESCRIPTION
**Pull request by jesseproudman**
Added instructions on how to use flipper-activerecord with activerecord 3.2.21 using the https://github.com/matthuhiggins/foreigner gem.

Still working through the finer details here.

### CODE REVIEW

**Comment by jesseproudman** [comment]
Well, this isn't super pretty to make it compatible with Rails 3.2, but it is working. Would love any thoughts / comments?

**Comment by bgentry** [comment]
@blueboxjesse wasn't watching this repo, sorry I just saw this. I don't have time to look through the PR now but I'll get to it soon. Feel free to ping me if I haven't reviewed in the next couple days :)

**Comment by jesseproudman** [comment]
Hey, no worries. We're using my forked version now in production - Let me know if I can answer any questions.

File: lib/flipper/adapters/activerecord.rb [hunk 1]
```diff
@@ -24,7 +24,9 @@ def features
 
       # Public: Adds a feature to the set of known features.
       def add(feature)
-        Flipper::ActiveRecord::Feature.find_or_create_by!(name: feature.name.to_s)
+        opts = { name: feature.name.to_s }
+        Flipper::ActiveRecord::Feature.where(opts).first ||
+          Flipper::ActiveRecord::Feature.create!(opts)
```
Comment (bgentry) [inline lib/flipper/adapters/activerecord.rb:29]: Can you share the reasoning behind this change?

File: lib/flipper/adapters/activerecord.rb [hunk 2]
```diff
@@ -44,7 +46,7 @@ def clear(feature)
       # Returns a Hash of Flipper::Gate#key => value.
       def get(feature)
         result = {}
-        f = Flipper::ActiveRecord::Feature.eager_load(:gates).find_by(name: feature.key)
+        f = Flipper::ActiveRecord::Feature.eager_load(:gates).where(name: feature.key).first
```
Comment (bgentry) [inline lib/flipper/adapters/activerecord.rb:49]: Same question on this change. Aren't these lines functionally equivalent?

**Comment by bgentry** [comment]
oh, crap, I see that most of these changes were made because the finder methods I used are new to Rails 4 and therefore you can't use them with Rails 3. That's unfortunate, as it seems to dirty up the code a fair bit :(

My inclination is to let you keep Rails 3.2 compatibility in your fork. I'd be willing to merge any changes necessary for 4.X support (removing the foreign key and lowering AR version requirements), but I'd rather not merge in changes that use more verbose or less clear ActiveRecord queries just for the sake of compatibility with a no-longer-supported version of Rails.

If you'd like, you can submit a PR to add a readme change that links to your fork for those wanting Rails 3.2 support.

Sorry this took me so long to review!

**Comment by jesseproudman** [comment]
No problem. Makes total sense. Opened a new PR here: https://github.com/bgentry/flipper-activerecord/pull/3

**Pull request closed without merging**